syntax = "proto3";

package users;

option java_package = "com.empuje.grpc.users";
option java_multiple_files = true;

// No exponemos password en las respuestas.
message User {
  int64 id = 1;
  string username = 2;
  string nombre = 3;
  string apellido = 4;
  string telefono = 5;
  string email = 6;
  int64 rol_id = 7;
  bool activo = 8;
}

message CreateUserRequest {
  string username = 1;
  string nombre = 2;
  string apellido = 3;
  string telefono = 4;
  string email = 5;
  string password = 6; // Sólo entra en request
  int64 rol_id = 7;    // opcional (por defecto VOLUNTARIO = 4)
}

message CreateUserResponse {
  User user = 1;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool ok = 1;
  string mensaje = 2;
  User user = 3; // si ok=true, devolvemos datos públicos
}

message GetUserRequest {
  int64 id = 1;
}
message GetUserResponse {
  User user = 1;
}

message ListUsersRequest {
  // filtros simples opcionales
  bool only_active = 1;
  int64 rol_id = 2;
}
message ListUsersResponse {
  repeated User users = 1;
}

service UsersService {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc GetUser (GetUserRequest) returns (GetUserResponse);
  rpc ListUsers (ListUsersRequest) returns (ListUsersResponse);
}
